tools/test_full_sync.py

```diff
@@ -16,7 +19,41 @@
 from chia.cmds.init_funcs import chia_init
 
 
-async def run_sync_test(file: Path, db_version=2) -> None:
+exit_with_failure = False
+
+
+class ExitOnError(logging.Handler):
+    def handle(self, record):
+        pass
+
+    def emit(self, record):
+        if record.level != logging.ERROR:
+            return
+        global exit_with_failure
+        exit_with_failure = True
+
+    def createLock(self):
+        self.lock = None
+
+    def _at_fork_reinit(self):
+        pass
```

Comment:
So I generally don't encourage inheriting, and it might be the best way here given the existing `Handler` code, but you already are anyways and could just inherit from `NullHandler` itself and drop three methods here.  Or, if keeping this structure, I would add a link to the reference code.

https://github.com/python/cpython/blob/v3.10.2/Lib/logging/__init__.py#L2203-L2223